src/core/service.c

```diff
@@ -4878,6 +4921,11 @@ static void service_notify_message_process_state(Service *s, char * const *tags)
                     s->state == SERVICE_START)
                         service_enter_start_post(s);
 
+                /* Check for signal handler only on state transition, not for repeated READY=1 notifications */
+                if (s->type == SERVICE_NOTIFY_RELOAD &&
+                    !IN_SET(old_notify_state, NOTIFY_READY, NOTIFY_RELOAD_READY))
+                        (void) service_check_reload_signal_handler(s, RELOAD_SIGNAL_HANDLER_WARN);
```

Comment:
I'd actually conditionalize this on SERVICE_START state, not previous notify_state